### PR TITLE
Update conformance

### PR DESCRIPTION
**Pull request by adrianhopebailie**
Added the same conformance text used in the Payment request API for consistency

### CODE REVIEW

**Comment by ianbjacobs** [comment]
Hi Adrian,

I have some comments on this PR. 
- " A conforming Payment Request API user agent MUST also be a   <em>conforming implementation</em> of the IDL fragments of this specification, as described in the “Web IDL” specification. [[!WEBIDL]]". (But see below for an alternative.)

Your text suggests an interesting point, but I'm not sure I would write it that way. Here is what I understand:
- Because this spec builds on payment request API, conformance to this specification also requires conformance to that specification.
- Conformance to this specification includes conformance to the WebIDL parts.
- "This specification uses both the terms "conforming user agent(s)" and "user agent(s)" to refer to this product class." It seems we should add "browser" to that list because that's the one we use most commonly.

-"In this specification, <a>user agent</a> means a <em>Web browser or other interactive user agent</em> as defined in [[!HTML5]]."  Are you sure you want to constrain the document that way? I'm ok to do so but want to be sure you have that in mind. I know you are thinking that some payment apps might not be interactive, for example. So I want to be sure you also don't have in mind non-interactive user agents.
- "A <dfn data-lt="user agents">user agent</dfn> MUST behave as described in this specification in order to be considered conformant." I suggest instead:
  "A conforming user agent is one that implements all of the "MUST" statements and the WebIDL."

In other words:
- We don't "consider" something to be conformant. It either fulfills requirements for conformance or it doesn't.
- The requirements are expressed through MUST statements.

**Comment by adrianhopebailie** [comment]
This is a copy paste from the Payment Request API spec for consistency. As discussed, the intent with this proposal is to have a partner spec to Payment Request with the same scope and target implementors. 

As such I think I created some confusion by not fixing _A conforming Payment Request API user agent_ which should read _A conforming Payment App API user agent _.

Fixed.

> It seems we should add "browser" to that list because that's the one we use most commonly.

I think for consistency we should use user agent and rather go through the doc and change any instances of browser to that. 

Fixed.

> Are you sure you want to constrain the document that way?

Yes. I am proposing that this document and Payment Request be absolutely consistent. If we want to change this language we should do so in both specs.

**Comment by adrianhopebailie** [comment]
I have merged this so that the many small changes (replace browser with user agent) don't make it difficult to review other PRs.

Can revert if required.
